expression/builtin_compare.go

```diff
@@ -963,6 +1034,17 @@ func (b *builtinLeastTimeSig) evalTime(row chunk.Row) (res types.Time, isNull bo
 			res = v
 		}
 	}
+	// Convert ETType Time value to MySQL actual type, distinguish date and datetime
+	sc := b.ctx.GetSessionVars().StmtCtx
+	var resTimeTp byte
+	if b.cmpAsDate {
+		resTimeTp = mysql.TypeDate
```

Comment:
Why not use `b.tp` to as resTimeTp?

Reply:
Currently, b.tp is generated from EType, which ignores the differences between date and datetime.